### PR TITLE
Image serializer

### PR DESCRIPTION
**Pull request by meztez**
Suggestion to fix issue #517. 

Modified images handling to serialize during serialization step instead of postexec step

Comments welcome.

 PR task list:
 
 * [x]  Update NEWS
 * [ ]  Add tests
 * [ ]  Update documentation with `devtools::document()`

### CODE REVIEW

**Comment by schloerke** [comment]
Seems odd to set the `res$serializer` in a `post` key val.  

Note: Look into `pre`/`post` and what they mean

**Comment by meztez** [comment]
I agree, image handling seems to be its own thing. Maybe it could be refactored to match the rest of the components?

**Comment by meztez** [comment]
it is preexec and postexec, I was mistaken.

File: R/images.R [hunk 1]
```diff
@@ -2,32 +2,31 @@
...
-      res
+      unlink(data$file)
+      img
     }
   )
 }
 
 render_jpeg <- function(args){
-  render_image(jpeg, "image/jpeg", args)
+  render_image(jpeg, args)
```
Comment (cpsievert) [inline R/images.R:27]:
```suggestion
  render_image(grDevices::jpeg, args)
```

File: R/images.R [hunk 2]
```diff
@@ -2,32 +2,31 @@
...
 }
 
 render_png <- function(args){
-  render_image(png, "image/png", args)
+  render_image(png, args)
```
Comment (cpsievert) [inline R/images.R:31]:
```suggestion
  render_image(grDevices::png, args)
```

File: R/images.R [hunk 3]
```diff
@@ -2,32 +2,31 @@
...
+    postexec = function(value, req, res, data){
       dev.off()
 
       con <- file(data$file, "rb")
       img <- readBin(con, "raw", file.info(data$file)$size)
       close(con)
-      res$body <- img
-      res$setHeader("Content-type", contentType)
-      res
+      unlink(data$file)
```
Comment (cpsievert) [inline R/images.R:20]: Consider doing this in an on.exit() before reading (in case it fails)

**Pull request merged**